craft_cli/messages.py

```diff
@@ -633,15 +639,28 @@ def stop(self) -> None:
 class _StreamContextManager:
     """A context manager that provides a pipe for subprocess to write its output."""
 
-    def __init__(
-        self, printer: _Printer, text: str, stream: Optional[TextIO], use_timestamp: bool
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        printer: _Printer,
+        text: str,
+        stream: Optional[TextIO],
+        use_timestamp: bool,
+        ephemeral_mode: bool,
     ):
+        # prepare the printer flags for the initial message and everything produced
+        # by the pipe reader
+        printer_flags = {
+            "use_timestamp": use_timestamp,
+            "ephemeral": ephemeral_mode,
+            "end_line": not ephemeral_mode
```

Comment:
Oops, that linting error was my fault. Sorry!
```suggestion
            "end_line": not ephemeral_mode,
```

Reply:
No worries, fixed.